be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1957,7 +1957,10 @@ void SegmentIterator::_replace_version_col(size_t num_rows) {
 uint16_t SegmentIterator::_evaluate_vectorization_predicate(uint16_t* sel_rowid_idx,
                                                             uint16_t selected_size) {
     SCOPED_RAW_TIMER(&_opts.stats->vec_cond_ns);
-    if (_is_need_vec_eval) {
+    // The expr can only be skipped when _non_predicate_columns is not empty;
+    // otherwise, skipping the expr could result in an empty block
+    // being output while _non_predicate_columns is empty.
+    if (_is_need_vec_eval && !_non_predicate_columns.empty()) {
```

Comment:
why the output will be empty block? since we already read the predicate column into current_return_columns and line 1970 will set all flags to true.